activerecord/test/cases/tasks/database_tasks_test.rb

```diff
@@ -467,6 +467,17 @@ class DatabaseTasksStructureLoadTest < ActiveRecord::TestCase
   end
 
   class DatabaseTasksCheckSchemaFileTest < ActiveRecord::TestCase
+    def setup
+      unless Rails.respond_to?(:root)
+        @defined_root = true
+        def Rails.root; :defined_root; end
```

Comment:
Minitest's `stub` only works if the method already exists, which isn't the case here.